File: web/src/services/todo.ts
import axios from "axios";

const URL = "http://localhost:8080/todos";

export async function update(item: any) {
  return await axios.patch(`${URL}/` + item.id, {
    description: item.description,
  });
}

export async function list() {
  return await axios.get(URL);
}

export async function remove(id: string) {
  return await axios.delete(`${URL}/` + id);
}

export async function get(id: string) {
  return await axios.get(`${URL}/` + id);
}

export async function create(description: string) {
  return await axios.post(URL, { description });
}

File: api/src/routes/todo.rs
use axum::routing::{delete, get};
use axum::Router;

use crate::models::todo::{create, delete, get, list, update};
use crate::global_structs::app_state::AppState;

pub fn routes(state: AppState) -> Router {
    Router::new()
        .route("/todos", get(list::handler).post(create::handler))
        .route(
            "/todos/:id",
            delete(delete::handler)
                .patch(update::handler)
                .get(get::handler),
        )
        .with_state(state)
}
